server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -391,6 +392,20 @@ public NodeEnvironment(Settings settings, Environment environment, IndexStoreLis
                 ensureNoFileCacheData(fileCacheNodePath);
             }
 
+            if (DiscoveryNode.isOfflineNode(settings) && DiscoveryNode.getRolesFromSettings(settings).size() > 1) {
```

Comment:
Please could you help add tests for this?

Reply:
tests are pending. will be working on adding those